src/mp/util.cpp

```diff
@@ -37,14 +39,17 @@ size_t MaxFd()
 
 std::string ThreadName(const char* exe_name)
 {
-    char thread_name[17] = {0};
+    char thread_name[16] = {0};
+#ifdef HAVE_PTHREAD_GETNAME_NP
     pthread_getname_np(pthread_self(), thread_name, sizeof(thread_name));
-    uint64_t tid = 0;
-#if __linux__
-    tid = syscall(SYS_gettid);
 #else
-    pthread_threadid_np(NULL, &tid);
-#endif
+    // By default, all the threads inherit the program name. So use that one
+    // if we don't have pthread_getname_np().
+    snprintf(thread_name, sizeof(thread_name), "%s", exe_name);
```

Comment:
I think this isn't safe because exe_name can be null here. Probably better in any case just to leave thread_name empty if it's not known, so exe_name isn't repeated twice below.

Reply:
Removed. In this case we would return something like `prog-35291/-20080` which looks a bit strange and may break some tool that tries to parse it.

Reply:
re: https://github.com/chaincodelabs/libmultiprocess/pull/21#discussion_r368248257

> Removed. In this case we would return something like `prog-35291/-20080` which looks a bit strange and may break some tool that tries to parse it.

I'm not too concerned about appearance of the dash, but omitting it when the thread name is empty could be a possible improvement

Reply:
https://github.com/chaincodelabs/libmultiprocess/pull/24 Don't print a dash if thread name is not known